src/mixins/java/org/spongepowered/common/mixin/core/world/level/ExplosionMixin.java

```diff
@@ -118,158 +113,107 @@ public abstract class ExplosionMixin implements ExplosionBridge {
-                            d1 = d1 / d3;
-                            d2 = d2 / d3;
-                            float f = this.radius * (0.7F + this.level.random.nextFloat() * 0.6F);
-                            double d4 = this.x;
-                            double d6 = this.y;
-                            double d8 = this.z;
-
-                            for (final float f1 = 0.3F; f > 0.0F; f -= 0.22500001F) {
-                                final BlockPos blockpos = new BlockPos(d4, d6, d8);
-                                final BlockState blockstate = this.level.getBlockState(blockpos);
-                                final FluidState fluidstate = this.level.getFluidState(blockpos);
-                                Optional<Float> optional = this.damageCalculator.getBlockExplosionResistance((net.minecraft.world.level.Explosion) (Object) this, this.level, blockpos, blockstate, fluidstate);
-                                if (optional.isPresent()) {
-                                    f -= (optional.get() + 0.3F) * 0.3F;
-                                }
-
-                                if (f > 0.0F && this.damageCalculator.shouldBlockExplode((net.minecraft.world.level.Explosion) (Object) this, this.level, blockpos, blockstate, f)) {
-                                    set.add(blockpos);
-                                }
-
-                                d4 += d0 * (double) 0.3F;
-                                d6 += d1 * (double) 0.3F;
-                                d8 += d2 * (double) 0.3F;
-                            }
-                        }
-                    }
-                }
-            }
-
-            this.toBlow.addAll(set);
-        }
-        // Sponge End
+    }
 
-        final float f3 = this.radius * 2.0F;
-        final int k1 = Mth.floor(this.x - (double) f3 - 1.0D);
-        final int l1 = Mth.floor(this.x + (double) f3 + 1.0D);
-        final int i2 = Mth.floor(this.y - (double) f3 - 1.0D);
-        final int i1 = Mth.floor(this.y + (double) f3 + 1.0D);
-        final int j2 = Mth.floor(this.z - (double) f3 - 1.0D);
-        final int j1 = Mth.floor(this.z + (double) f3 + 1.0D);
+    @ModifyConstant(method = "explode", constant = @Constant(intValue = 16, ordinal = 2))
+    public int impl$explode_skipLoop(int previousValue) {
+        return this.impl$shouldBreakBlocks ? previousValue : 0;
+    }
 
-        // Sponge Start - Only query for entities if we're to damage them
-        final List<Entity> list = this.impl$shouldDamageEntities ? this.level.getEntities(this.source,
-                new AABB((double) k1, (double) i2, (double) j2, (double) l1, (double) i1, (double) j1)) : Collections.emptyList();
-        // Sponge End
+    @Redirect(
+            method = "explode", at = @At(
+            value = "INVOKE",
+            target = "Lnet/minecraft/world/level/Level;getEntities(Lnet/minecraft/world/entity/Entity;Lnet/minecraft/world/phys/AABB;)Ljava/util/List;"
+    ))
```

Comment:
```suggestion
    @Redirect(
        method = "explode",
        at = @At(
            value = "INVOKE",
            target = "Lnet/minecraft/world/level/Level;getEntities(Lnet/minecraft/world/entity/Entity;Lnet/minecraft/world/phys/AABB;)Ljava/util/List;"
        )
    )
```
Keep the spacing consistent.